src/meshes/tetgen_mesh.template.h

```diff
@@ -609,8 +627,20 @@ namespace oomph
               tetgen_io.regionlist[counter] =
                 static_cast<double>(srf_pt->region_id_for_tetgen(j));
               ++counter;
-              // Area target
-              tetgen_io.regionlist[counter] = 0.0;
+
+              // if there's no target volumes specified, default to zero
+              if (target_element_volume_in_region_pt == nullptr)
+              {
+                tetgen_io.regionlist[counter] = 0;
+              }
+              else
+              {
+                // deliberate integer division here to round down to the region
+                // number (five doubles per region)
+                tetgen_io.regionlist[counter] =
+                  (*target_element_volume_in_region_pt)[counter / 5];
```

Comment:
Up for discussion but I'd much prefer using something like
```cpp
(*target_element_volume_in_region_pt)[int(counter / 5)];
```
for the argument. It's then immediately clear from the code that the integer division is explicit (rather than me needing to read the comment above it to do that).

Reply:
agreed, changed this to unsigned(counter/5) in the updated commit